rpc/src/rpc_auth/mod.rs

```diff
@@ -117,10 +118,19 @@ impl<B> ValidateRequest<B> for RpcAuth {
             Ok(true) => Ok(()),
             Ok(false) => {
                 log::error!("Unauthorized RPC request {:?}", request.uri());
+                let status = http::StatusCode::UNAUTHORIZED;
+                let err_obj = ErrorObject::owned(
+                    status.as_u16().into(),
+                    status.canonical_reason().unwrap_or_default(),
+                    None::<()>,
+                );
+                let body = serde_json::to_string(&err_obj).expect("constant object");
+
                 Err(Response::builder()
-                    .status(http::StatusCode::UNAUTHORIZED)
+                    .status(status)
                     .header(http::header::WWW_AUTHENTICATE, "Basic")
-                    .body(Body::empty())
+                    .header(http::header::CONTENT_TYPE, "application/json")
+                    .body(body.into())
                     .expect("must be valid"))
             }
```

Comment:
This doesn't conform to the correct format. Not that I'm an expert in this, but I checked online. This is the correct format for the error object, but not the json response. Trying this out:

```
curl -H 'Content-Type: application/json'   -d '{"jsonrpc": "2.0", "id": 1, "method": "chainstate_info", "params": []}' http://127.0.0.1:3030
{"code":401,"message":"Unauthorized"}
```

But the part `{"code":401,"message":"Unauthorized"}` should be inside a json RPC response, so something like this:

```
{
  "jsonrpc": "2.0",
  "error": {
    "code": 401,
    "message": "Unauthorized",
  },
  "id": 1
}
```

Reply:
Changed. The `id` field is null though, since authentication happens before request ID is available but I think this is correct according to the spec.